lib/rubocop/cop/github/insecure_hash_algorithm.rb

```diff
@@ -6,33 +6,89 @@ module RuboCop
   module Cop
     module GitHub
       class InsecureHashAlgorithm < Cop
-        MSG = "This hash algorithm is old and insecure, use SHA-256 instead"
+        MSG = "This hash function is not allowed"
 
+        # Matches constants like these:
+        #   Digest::MD5
+        #   OpenSSL::Digest::MD5
         def_node_matcher :insecure_const?, <<-PATTERN
           (const (const _ :Digest) #insecure_algorithm?)
         PATTERN
 
-        def_node_matcher :insecure_call?, <<-PATTERN
-          (send (const _ {:Digest :HMAC}) _ (str #insecure_algorithm?) ...)
+        # Matches calls like these:
+        #   Digest.new('md5')
+        #   Digest.hexdigest('md5', 'str')
+        #   OpenSSL::Digest.new('md5')
+        #   OpenSSL::Digest.hexdigest('md5', 'str')
+        #   OpenSSL::Digest::Digest.new('md5')
+        #   OpenSSL::Digest::Digest.hexdigest('md5', 'str')
+        #   OpenSSL::Digest::Digest.new(:MD5)
+        #   OpenSSL::Digest::Digest.hexdigest(:MD5, 'str')
+        def_node_matcher :insecure_digest?, <<-PATTERN
+          (send
+            (const _ {:Digest :HMAC})
+            #not_just_encoding?
+            #insecure_algorithm?
+            ...)
+        PATTERN
+
+        # Matches calls like "Digest(:MD5)".
+        def_node_matcher :insecure_hash_lookup?, <<-PATTERN
+          (send _ :Digest #insecure_algorithm?)
         PATTERN
 
         def insecure_algorithm?(val)
-          case val.to_s.downcase
-          when "md5", "sha1"
-            true
-          else
+          return false if val == :Digest # Don't match "Digest::Digest".
+          case str_val(val).downcase
+          when *allowed_hash_functions
             false
+          else
+            true
           end
         end
 
+        def not_just_encoding?(val)
+          !just_encoding?(val)
+        end
+
+        def just_encoding?(val)
+          val == :hexencode || val == :bubblebabble
+        end
+
+        # Built-in hash functions are listed in these docs:
+        #  https://ruby-doc.org/stdlib-2.7.0/libdoc/digest/rdoc/Digest.html
+        #  https://ruby-doc.org/stdlib-2.7.0/libdoc/openssl/rdoc/OpenSSL/Digest.html
+        DEFAULT_ALLOWED = %w[
+          RIPEMD160
+          RMD160
```

Comment:
These shouldn't be allowed.

Reply:
I removed them (and SHA-224). The default allow list now is:

https://github.com/github/rubocop-github/blob/4d952bbb1da770bfe49586a4db2a79061eed612a/lib/rubocop/cop/github/insecure_hash_algorithm.rb#L61-L65